cmd/root.go

```diff
@@ -299,6 +299,17 @@ func rootPersistentPreRunEFunc(f *factory.Factory, cfg *configuration.Config) fu
 		if err := configuration.CheckMinAPIVersionRestriction(cmd, cfg.Version); err != nil {
 			return err
 		}
+
+		// Check for new sdpctl version
+		meta, err := cfg.CheckForUpdate(f.StdErr, version)
+		if err != nil {
+			log.WithError(err).Info("version check result")
+		}
+		viper.Set("meta", meta)
+		if err := viper.WriteConfig(); err != nil {
```

Comment:
refactor with https://github.com/appgate/sdpctl/blob/23e58b1d42673a64e6a4b9632b215895e25a73a1/cmd/root.go#L276-L280 so we only write once, and handle the case if we configure sdpctl with environment variables only without config file. I'm pretty sure the current version here would fail in in the e2e test